lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -359,6 +362,23 @@ function _campaign_resource_reportback($nid, $values) {
     watchdog('dosomething_api', '_campaign_resource_reportback:' . json_encode($values));
   }
 
-  return dosomething_reportback_save($values, $user);
+  $transaction_id = $_SERVER['HTTP_X_REQUEST_ID'];
+
+  if (! $transaction_id) {
+    $rogue_reportback = dosomething_rogue_send_reportback_to_rogue($values, $user);
+
+    // Make sure the reportback exists (meaning it got back from Rogue).
+    $rbid = dosomething_reportback_exists($rogue_reportback['data']['campaign_id'], $rogue_reportback['data']['campaign_run_id'], $rogue_reportback['data']['drupal_id']);
+
+    // Store reference to the rb in rogue.
+    if ($rbid) {
+      dosomething_rogue_store_rogue_references($rbid, $rogue_reportback);
```

Comment:
I think we only want to call this function if there is a new image as well, since one of the columns is `fid` (and we also pop the most recent `fid` so if there was no new file we'd end up with one that should already be in there). I wonder if this is what was causing those `fid` related errors before?